From a88230244fdd6486af67828f324e2d6fbe3ab7f8 Mon Sep 17 00:00:00 2001
From: Dibyajyoti Pal <dibyapal@google.com>
Date: Mon, 10 Jun 2024 18:21:15 +0000
Subject: [PATCH] [PWA] Fix crash in WebAppIconDiagnosticCommand

This CL fixes a crash in WebAppIconDiagnosticCommand with an inverted
if condition.

Bug: 346180350
Change-Id: Iff49a62e79d63a251645db009caa475d4275ab02
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5615892
Auto-Submit: Dibyajyoti Pal <dibyapal@chromium.org>
Reviewed-by: Daniel Murphy <dmurph@chromium.org>
Commit-Queue: Daniel Murphy <dmurph@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1312925}
---

diff --git a/chrome/browser/web_applications/commands/web_app_icon_diagnostic_command.cc b/chrome/browser/web_applications/commands/web_app_icon_diagnostic_command.cc
index cab2363..8c71d4e 100644
--- a/chrome/browser/web_applications/commands/web_app_icon_diagnostic_command.cc
+++ b/chrome/browser/web_applications/commands/web_app_icon_diagnostic_command.cc
@@ -21,7 +21,7 @@
 
 base::Value CreateIconDiagnosticDebugData(
     const std::optional<WebAppIconDiagnosticResult> diagnostic_result) {
-  if (diagnostic_result.has_value()) {
+  if (!diagnostic_result.has_value()) {
     return base::Value("no_result_yet");
   }
   base::Value::Dict root;
